frontend/packages/dev-console/src/utils/resource-label-utils.ts

```diff
@@ -90,8 +102,10 @@ export const mergeData = (originalResource: K8sResourceKind, newResource: K8sRes
   };
   if (mergedData.metadata.annotations) {
     mergedData.metadata.annotations = {
+      ...(isDevfileResource
+        ? originalResource?.metadata?.annotations
+        : getUserAnnotations(originalResource?.metadata?.annotations)),
       ...newResource.metadata.annotations,
-      ...(isDevfileResource ? originalResource?.metadata?.annotations : {}),
```

Comment:
Interesting how we went at this... but I think the movement to before is a good idea. Original sets it up, New overrides what it needs.